applications/GeoMechanicsApplication/custom_constitutive/coulomb_yield_surface.cpp

```diff
@@ -193,24 +176,61 @@ void CoulombYieldSurface::InitializeKappaDependentFunctions()
     mDilatancyAngleCalculator = MakeDilatancyAngleCalculator(mMaterialProperties);
 }
 
-double CoulombYieldSurface::CalculatePlasticMultiplier(const Vector& rSigmaTau, const Vector& rDerivativeOfFlowFunction)
+double CoulombYieldSurface::CalculatePlasticMultiplier(const Vector& rSigmaTau,
+                                                       const Vector& rDerivativeOfFlowFunction) const
 {
     const auto sin_phi   = std::sin(GetFrictionAngleInRadians());
     const auto numerator = sin_phi * rDerivativeOfFlowFunction[0] + rDerivativeOfFlowFunction[1];
     return (GetCohesion() * std::cos(GetFrictionAngleInRadians()) - rSigmaTau[0] * sin_phi - rSigmaTau[1]) / numerator;
 }
 
-double CoulombYieldSurface::CalculateEquivalentPlasticStrain(const Vector&        rSigmaTau,
-                                                             CoulombAveragingType AveragingType,
-                                                             double               lambda) const
-{
-    Vector     dGdsigma = DerivativeOfFlowFunction(rSigmaTau, AveragingType);
-    const auto g1       = (dGdsigma[0] + dGdsigma[1]) * 0.5;
-    const auto g3       = (dGdsigma[0] - dGdsigma[1]) * 0.5;
-    const auto mean     = (g1 + g3) / 3.0;
-    const auto deviatoric = std::sqrt(std::pow(g1 - mean, 2) + std::pow(g3 - mean, 2) + std::pow(mean, 2));
-    const auto alpha = std::sqrt(2.0 / 3.0) * deviatoric;
-    return -alpha * lambda;
+double CoulombYieldSurface::CalculateEquivalentPlasticStrainIncrement(const Vector& rSigmaTau,
+                                                                      CoulombAveragingType AveragingType) const
+{
+    const auto derivative              = DerivativeOfFlowFunction(rSigmaTau, AveragingType);
+    const auto principal_stress_vector = UblasUtilities::CreateVector(
+        {(derivative[0] + derivative[1]) / 2.0, 0.0, (derivative[0] - derivative[1]) / 2.0});
+    const auto mean =
+        std::accumulate(principal_stress_vector.begin(), principal_stress_vector.end(), 0.0) / 3.0;
```

Comment:
We could consider to replace the hard-coded size by asking for the vector's size:
```suggestion
    const auto mean = std::accumulate(principal_stress_vector.begin(), principal_stress_vector.end(), 0.0) /
                      static_cast<double>(principal_stress_vector.size());
```
What do you think? And perhaps @WPK4FEM may also want to weigh in?